packages/app/src/background/services/credentials/utils.ts

```diff
@@ -0,0 +1,243 @@
+    if (typeof value === "string") {
+      return true;
+    }
+
+    if (Array.isArray(value)) {
+      const validationResults = await Promise.all(
+        value.map(async (item) => {
+          try {
+            await claimValueSchema.validate(item);
+            return true;
+          } catch (error) {
+            return false;
+          }
+        }),
+      );
+      return validationResults.every((result) => result);
+    }
+
+    try {
+      await claimValueMapSchema.validate(value);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  });
+
+const credentialSubjectSchema: yup.Schema<CredentialSubject> = yup.object({
+  id: yup.string().strict().optional(),
+  claims: claimValueMapSchema,
+});
+
+const credentialIssuerSchema: yup.Schema<CredentialIssuer> = yup.object({
+  id: yup.string().strict().optional(),
+});
+
+const credentialProofSchema: yup.Schema<CredentialProof> = yup.object({
+  id: yup.string().strict().optional(),
+  type: yup.array().strict().of(yup.string().required()).required(),
+  proofPurpose: yup.string().strict().required(),
+  verificationMethod: yup.string().strict().required(),
+  created: yup.date().transform(parseDate).required(),
+  proofValue: yup.string().strict().required(),
+});
+
+const credentialStatusSchema: yup.Schema<CredentialStatus> = yup.object({
+  id: yup.string().strict().required(),
+  type: yup.string().strict().required(),
+});
+
+const verifiableCredentialSchema: yup.Schema<VerifiableCredential> = yup.object({
+  context: yup.array().strict().of(yup.string().required()).required(),
+  id: yup.string().strict().optional(),
+  type: yup.array().strict().of(yup.string().required()).required(),
+  issuer: yup
+    .mixed()
+    .required()
+    .test("is-string-or-CredentialIssuer", "issuer must be a string or CredentialIssuer", async (value) => {
+      if (typeof value === "string") {
+        return true;
+      }
+
+      try {
+        await credentialIssuerSchema.validate(value);
+        return true;
+      } catch (error) {
+        return false;
+      }
+    }),
+  issuanceDate: yup.date().transform(parseDate).required(),
+  expirationDate: yup.date().transform(parseDate).optional(),
+  credentialSubject: credentialSubjectSchema,
+  credentialStatus: yup.lazy((value) => {
+    if (!value) {
+      return yup.object().optional();
+    }
+
+    return credentialStatusSchema.required() as yup.Schema;
+  }),
+  proof: yup.array().of(credentialProofSchema).optional(),
+});
+
+/**
+ * Determines if an object is a valid VerifiableCredential.
+ * @param verifiableCredential An object representing a VerifiableCredential.
+ * @returns The Verifiable Credential if the object is a valid VerifiableCredential, otherwise throws an error.
+ */
+export async function validateVerifiableCredential(verifiableCredential: object): Promise<VerifiableCredential> {
+  try {
+    return await verifiableCredentialSchema.validate(verifiableCredential);
```

Comment:
```suggestion
    return verifiableCredentialSchema.validate(verifiableCredential);
```

Reply:
This await waits for the promise to resolve so that the error will actually be caught by the try-catch

Reply:
Ok, in this case you can remove try/catch. The error will be rejected anyway.

Reply:
Ah oops, good catch